### PR TITLE
Clean up deprecation warnings for utcnow and utcfromtimestamp

### PR DESCRIPTION
**Pull request by rafalkrupinski**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -859,7 +859,7 @@ import datetime
 from httpx_auth.testing import browser_mock, BrowserMock, create_token
 
 def test_something(browser_mock: BrowserMock):
-    token_expiry = datetime.datetime.utcnow() + datetime.timedelta(hours=1)
+    token_expiry = datetime.datetime.now(datetime.UTC) + datetime.timedelta(hours=1)
```
[the remark this answers is not in the document]
Comment (rafalkrupinski) [reply README.md:862]: hmm, I intended timezone.utc here bc I thought datetime.UTC is 3.11+, but actually tests on 3.9 pass...

[the remark this answers is not in the document]
Comment (Colin-b) [reply README.md:862]: That's because this is the README, tests are not executed against this sample :)

[the remark this answers is not in the document]
Comment (rafalkrupinski) [reply README.md:862]: yep, that might be it

File: httpx_auth/oauth2_tokens.py [hunk 2]
```diff
@@ -88,7 +87,7 @@ def _add_token(self, key: str, token: str, expiry: float):
             self.tokens[key] = token, expiry
             self._save_tokens()
             logger.debug(
-                f'Inserting token expiring on {datetime.datetime.utcfromtimestamp(expiry)} (UTC) with "{key}" key: {token}'
+                f'Inserting token expiring on {datetime.datetime.fromtimestamp(expiry, datetime.timezone.utc)} (UTC) with "{key}" key: {token}'
```
[the remark this answers is not in the document]
Comment (rafalkrupinski) [reply httpx_auth/oauth2_tokens.py:90]:
```python
>>> f'{datetime.datetime.fromtimestamp(expiry, datetime.timezone.utc)}'
'2024-01-23 18:51:39.843675+00:00'
```

I guess it makes sense to remove UTC part then

File: httpx_auth/oauth2_tokens.py [hunk 3]
```diff
@@ -122,7 +121,7 @@ def get_token(
                     del self.tokens[key]
                 else:
                     logger.debug(
-                        f"Using already received authentication, will expire on {datetime.datetime.utcfromtimestamp(expiry)} (UTC)."
+                        f"Using already received authentication, will expire on {datetime.datetime.fromtimestamp(expiry, datetime.timezone.utc)} (UTC)."
```
Comment (Colin-b) [inline httpx_auth/oauth2_tokens.py:124]:
```suggestion
                        f"Using already received authentication, will expire on {datetime.datetime.fromtimestamp(expiry, datetime.timezone.utc)}."
```

File: httpx_auth/oauth2_tokens.py [hunk 4]
```diff
@@ -144,7 +143,7 @@ def get_token(
                 if state in self.tokens:
                     bearer, expiry = self.tokens[state]
                     logger.debug(
-                        f"Using newly received authentication, expiring on {datetime.datetime.utcfromtimestamp(expiry)} (UTC)."
+                        f"Using newly received authentication, expiring on {datetime.datetime.fromtimestamp(expiry, datetime.timezone.utc)} (UTC)."
```
Comment (Colin-b) [inline httpx_auth/oauth2_tokens.py:146]:
```suggestion
                        f"Using newly received authentication, expiring on {datetime.datetime.fromtimestamp(expiry, datetime.timezone.utc)}."
```

File: httpx_auth/oauth2_tokens.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (Colin-b) [reply httpx_auth/oauth2_tokens.py]: The 2 other places should also be updated (see proposed changes)

**Comment by rafalkrupinski** [comment]
Squashed and split the messages into shorter lines

**Review by Colin-b — APPROVED** [review]

**Pull request merged**